pkg/quota/openstack/limits.go

```diff
@@ -0,0 +1,101 @@
+package openstack
+
+import (
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack"
+	computequotasets "github.com/gophercloud/gophercloud/openstack/compute/v2/extensions/quotasets"
+	tokensv2 "github.com/gophercloud/gophercloud/openstack/identity/v2/tokens"
+	tokensv3 "github.com/gophercloud/gophercloud/openstack/identity/v3/tokens"
+	"github.com/gophercloud/utils/openstack/clientconfig"
+	"github.com/pkg/errors"
+)
+
+// record stores the data from quota limits and usages.
+type record struct {
+	Service string
+	Name    string
+	Value   int64
+}
+
+// loadLimits loads the consumer quota metric.
+func loadLimits(opts *clientconfig.ClientOpts) ([]record, error) {
+	var limits []record
+
+	endpointOpts := gophercloud.EndpointOpts{}
+	authOpts, err := clientconfig.AuthOptions(opts)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get client auth options")
+	}
+
+	providerClient, err := openstack.AuthenticatedClient(*authOpts)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get keystone token")
+	}
+
+	projectID, err := getProjectID(providerClient)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get keystone project ID")
+	}
+
+	computeRecords, err := getComputeLimits(providerClient, endpointOpts, projectID)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get compute quota records")
+	}
+	for _, r := range computeRecords {
+		limits = append(limits, r)
+	}
+
+	return limits, nil
+}
+
+func getComputeLimits(pClient *gophercloud.ProviderClient, eo gophercloud.EndpointOpts, projectID string) ([]record, error) {
+	computeClient, err := openstack.NewComputeV2(pClient, eo)
```

Comment:
If you need to create an openstack client, try to use the `NewServiceClient()` function instead as it will make it easier to [support self-signed certificates](https://github.com/openshift/installer/pull/4405).

Reply:
ack, and just to make sure the rest is fine, can you confirm that we don't need to call `NewServiceClient()` for:

https://github.com/openshift/installer/blob/9665db1d1783ff540e9631501ea39e54490eb265/pkg/quota/openstack/limits.go#L77